RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -1097,9 +1111,13 @@ void MuonIdProducer::fillArbitrationInfo( reco::MuonCollection* pOutputMuons )
          for ( auto& chamber : muon1.matches() )
          {
            if(!(chamber.station()==stationIndex && chamber.detector()==detectorIndex)) continue;
-           if(chamber.segmentMatches.empty()) continue;
+           std::vector<reco::MuonSegmentMatch> * segmentMatches = nullptr;
+           if      (muonType == reco::Muon::TrackerMuon) segmentMatches =  & chamber.segmentMatches;
```

Comment:
this looks incomplete: the detectorIndex loop goes over DT, CSC, and RPC and does not cover the ME0 and GEM types.
Change the detectorIndex range to include 4 and 5.